src/features/loading/utils.ts

```diff
@@ -108,7 +108,5 @@ export const checkHashAndDownloadParams = async ({
     promises.push(promise)
   }
 
-  for (const promise in promises) {
-    await promise
-  }
+  await Promise.all(promises)
```

Comment:
This will make to run the check for every file in parallel, making calls to onUpdate callback without order, resulting to status message on the loading screen to change chaotically

Reply:
No:
- First: `for (const promise in promises) {` the variable `promise` here is a string, because you were using `for ... in ...` you should use `for ... of ...` instead
- Second: At the time you initialize the promise in the above bigger `for` loop, it start checking/downloading immediately. So if you want that logic, you need to await inside the above bigger `for` loop like the old logic.

Reply:
@br4in3x Here I made an example for you:
- `forInLoop` is your logic, it would run in ~0ms because await on string is noop.
- `forOfLoop` is the fixed version, it would run in ~1000ms (the longest execution of all promises). It is equivalent to the `Promise.all` version.
- `oldForLoop` I guess it is what you want, which waits until the previous promise resolved then execute the next. It would run in ~2000ms

<img width="625" alt="Screen Shot 2021-04-26 at 02 07 20" src="https://user-images.githubusercontent.com/8970908/116006137-2984a080-a634-11eb-9c80-19bd5f0e5251.png">

Reply:
Addressed in https://github.com/pastelnetwork/pastel-electron-wallet/pull/37